AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.java

```diff
@@ -340,75 +315,22 @@ protected String getHumanReadableLoginErrorMessage(Exception exception) {
         return exception.getLocalizedMessage();
     }
 
-    public void handleCurrentOrientationState(int orientation) {
-        if ((orientation == Configuration.ORIENTATION_UNDEFINED) || orientation == Configuration.ORIENTATION_LANDSCAPE) {
-            mAnkidroidLogo.setVisibility(View.GONE);
-        } else {
-            mAnkidroidLogo.setVisibility(View.VISIBLE);
-        }
-    }
-
-    public void checkOrientation(int orientation) {
-        recordOrientation(orientation);
-        if (isScreenSmall() && isHorizontal(orientation)) {
-            mAnkidroidLogo.setVisibility(View.GONE);
-        } else {
-            mAnkidroidLogo.setVisibility(View.VISIBLE);
-        }
-    }
-
-    private boolean isHorizontal(int orientation) {
-        int avgOrientation = getAverageOrientation();
-        return Math.abs(orientation - avgOrientation) < DEVIATION
-                && ROTATED_RIGHT - OFFSET <= avgOrientation
-                && avgOrientation <= ROTATED_LEFT + OFFSET;
-    }
-
     private boolean isScreenSmall() {
         return (this.getApplicationContext().getResources().getConfiguration().screenLayout
                 & Configuration.SCREENLAYOUT_SIZE_MASK)
                 < Configuration.SCREENLAYOUT_SIZE_LARGE;
     }
 
-    private void recordOrientation(int orientation) {
-        if(orientation != OrientationEventListener.ORIENTATION_UNKNOWN) {
-            orientations[orientationIndex++ % ORIENTATION_WINDOW] = orientation;
-        }
-    }
-
-    private int getAverageOrientation() {
-        int orientationSum = 0;
-        for(int ort: orientations){
-            orientationSum += ort;
-        }
-        return orientationSum / orientations.length;
-    }
-
-    public void initializeOrientationListener() {
-        myOrientationEventListener = new OrientationEventListener(this, SensorManager.SENSOR_DELAY_NORMAL) {
-            @Override
-            public void onOrientationChanged(int orientation) {
-                checkOrientation(orientation);
-            }
-        };
-    }
-
     @Override
-    protected void onResume() {
-        super.onResume();
-        // handle current state
-        handleCurrentOrientationState(getResources().getConfiguration().orientation);
-        myOrientationEventListener.enable();
-    }
-
-
-    @Override
-    protected void onPause() {
-        super.onPause();
-        myOrientationEventListener.disable();
+    public void onConfigurationChanged(@NonNull Configuration newConfig) {
+        super.onConfigurationChanged(newConfig);
+        if(isScreenSmall() && newConfig.orientation == Configuration.ORIENTATION_LANDSCAPE){
```

Comment:
I'm not sure if `isScreenSmall()` will be valid here - `480x640` might not be sufficient.

It works on my screen, I'll do a little more exploratory testing.

```suggestion
        if (isScreenSmall() && newConfig.orientation == Configuration.ORIENTATION_LANDSCAPE) {
```

Reply:
Spacing... oops! I'll fix that and play with the emulators. And thank you for the feedback, and the recommendation to use onConfigurationChanged()!

Reply:
No worries! We have an extensive code style: https://github.com/ankidroid/Anki-Android/wiki/Code-style I'm working on automating things so we won't need to worry about it in reviews